enhancements/ingress/proxy-protocol.md

```diff
@@ -0,0 +1,193 @@
+
+When OpenShift is configured with an external load-balancer in front of an
+IngressController, the connections that the IngressController receives have the
+load balancer's address (instead of the original client addresses) as the source
+address.  PROXY protocol enables the external load-balancer to preserve the
+original client addresses and include them when forwarding connections to the
+IngressController, which can then use the original source addresses for
+filtering and injecting HTTP headers.
+
+### Goal
+
+1. Enable cluster administrators to enable PROXY protocol on IngressControllers that use the "HostNetwork" and "NodePortService" endpoint publishing strategy types.
+
+### Non-Goal
+
+1. Enabling PROXY protocol to be configured on individual Routes.
+
+## Proposal
+
+To enable cluster administrators to configure IngressControllers to use PROXY
+protocol, the IngressController API is extended by adding an optional
+`ProxyProtocol` field with type `ProxyProtocol` to the `HostNetworkStrategy` and
+`NodePortStrategy` structs:
+
+```go
+// HostNetworkStrategy holds parameters for the HostNetwork endpoint publishing
+// strategy.
+type HostNetworkStrategy struct {
+	ProxyProtocol ProxyProtocol `json:"proxyProtocol,omitempty"`
+}
+
+// NodePortStrategy holds parameters for the NodePortService endpoint publishing strategy.
+type NodePortStrategy struct {
+	ProxyProtocol ProxyProtocol `json:"proxyProtocol,omitempty"`
+}
+```
+
+`ProxyProtocol` is a string type with two allowed values:
+
+```go
+// ProxyProtocol specifies whether PROXY protocol is enabled or not.
+// +kubebuilder:validation:Enum=On;Off
+type ProxyProtocol string
+```
+
+The following example configures an IngressController that uses a NodePort
+service that communicates using PROXY protocol:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+metadata:
+  name: default
+  namespace: openshift-ingress-operator
+spec:
+  endpointPublishingStrategy:
+    type: NodePortService
+    nodePort:
+      proxyProtocol: On
+```
+
+#### Validation
+
+Specifying `spec.endpointPublishingStrategy.type: HostNetwork` and omitting
+`spec.endpointPublishingStrategy.hostNetwork` or
+`spec.endpointPublishingStrategy.hostNetwork.proxyProtocol` is valid and
+specifies the default behavior, which is not to use PROXY protocol.  Similarly,
+specifying `spec.endpointPublishingStrategy.type: NodePortService` and omitting
+`spec.endpointPublishingStrategy.nodePort` or
+`spec.endpointPublishingStrategy.nodePort.proxyProtocol` is valid and specifies
+the default behavior, which again is not to use PROXY protocol.  The API
+validates that any value provided for
+`spec.endpointPublishingStrategy.hostNetwork.proxyProtocol` or
+`spec.endpointPublishingStrategy.nodePort.proxyProtocol` is one of the allowed
+values: "On" or "Off".
+
+### User Stories
+
+#### As a cluster administrator, I need my IngressController to use PROXY protocol to communicate with my external load-balancer.
```

Comment:
I think you may need to drop the period at the end of this line to satisfy the new line rules.